pkg/controller/stacks/install/installjob.go

```diff
@@ -235,7 +235,7 @@ func (jc *stackInstallJobCompleter) createJobOutputObject(ctx context.Context, o
 	// when the current object is a Stack object, make sure the name and namespace are
 	// set to match the current StackInstall (if they haven't already been set). Also,
 	// set the owner reference of the Stack to be the StackInstall.
-	if isStackObject(obj) || isStackConfigurationObject(obj) {
+	if isStackObject(obj) || isStackDefinitionObject(obj) || isStackConfigurationObject(obj) {
```

Comment:
`handleJobCompletion` also needs some care to accommodate `stackdefinition` output.

Reply:
I've moved the special StackInstall updating from handleJobCompletion to the StackInstall sync function.